internal/pgdb/pgdb_field.go

```diff
@@ -373,5 +376,103 @@ func getCommonFields(ctx pgsgo.Context, m pgs.Message, ix *importTracker) ([]*fi
 		},
 		QueryTypeName: ctx.Name(m).String() + "PBData" + "QueryType",
 	}
-	return []*fieldContext{tenantIdField, pkskField, pkField, skField, ftsDataField, pbDataField}, nil
+
+	rv := []*fieldContext{tenantIdField, pkskField, pkField, skField, ftsDataField, pbDataField}
+
+	// iterate message for vector behavior options
+	for _, field := range m.Fields() {
+		ext := pgdb_v1.FieldOptions{}
+		_, err := field.Extension(pgdb_v1.E_Options, &ext)
+		if err != nil {
+			return nil, fmt.Errorf("pgdb: getField: failed to extract Message extension from '%s': %w", field.FullyQualifiedName(), err)
+		}
+		if ext.MessageBehavoir != pgdb_v1.FieldOptions_MESSAGE_BEHAVOIR_VECTOR {
+			continue
+		}
+		if !field.Type().IsRepeated() {
+			panic(fmt.Errorf("pgdb: vector behavior only supported on repeated fields: %s", field.FullyQualifiedName()))
+		}
+		subMsg := field.Type().Element().Embed()
+		if subMsg == nil {
+			panic(fmt.Errorf("pgdb: vector behavior only supported on message fields: %s", field.FullyQualifiedName()))
+		}
+		allFields := subMsg.Fields()
+		if len(allFields) != 2 {
+			panic(fmt.Errorf("pgdb: vector message must only have model enum and float array: %s", field.FullyQualifiedName()))
+		}
+
+		var enumField pgs.Field
+		var floatField pgs.Field
+		var vectorSize int32
+
+		for _, subField := range allFields {
+			switch subField.Descriptor().GetNumber() {
+			case 1:
+				// enum
+				if subField.Type().ProtoType() != pgs.EnumT {
+					panic(fmt.Errorf("pgdb: vector message must have model enum as first field: %s", field.FullyQualifiedName()))
+				}
+				enumField = subField
+			case 2:
+				// repeated float
+				if !subField.Type().IsRepeated() || subField.Type().Element().ProtoType() != pgs.FloatT {
+					panic(fmt.Errorf("pgdb: vector message must have repeated float as second field: %s", field.FullyQualifiedName()))
+				}
+				subExt := pgdb_v1.FieldOptions{}
+				_, err := subField.Extension(pgdb_v1.E_Options, &subExt)
+				if err != nil {
+					return nil, fmt.Errorf("pgdb: getField: failed to extract Message extension from '%s': %w", field.FullyQualifiedName(), err)
+				}
+				if subExt.VectorSize == 0 {
+					panic(fmt.Errorf("pgdb: vector message must have vector_size set on repeated float field: %s", field.FullyQualifiedName()))
+				}
+				vectorSize = subExt.VectorSize
+				floatField = subField
+			}
+		}
+
+		pgColName, err := getColumnName(field)
+		if err != nil {
+			panic(fmt.Errorf("pgdb: getColumnName failed for: %v: %s (of type %s)",
+				field.Type().ProtoType(), field.FullyQualifiedName(), field.Descriptor().GetType()))
+		}
+
+		// enum values
+		for _, enumValue := range enumField.Type().Enum().Values() {
+			if enumValue.Value() == 0 {
+				// skip the zero value
+				continue
+			}
+
+			goNameString := ctx.Name(field).String() + "_" + strings.TrimPrefix(ctx.Name(enumValue).String(), "MODEL_")
```

Comment:
The TrimPrefix here should be based on the `enumField` name

Reply:
enumField name is just "Model"

Reply:
Yeah, so, we should Trim the `strings.TrimPrefix(strings.ToUpper(enumField.SOMETHINGHERE) + "_")`?  not sure whats on the enumField to grab here

Reply:
what about `ModelEmbeddings_PastaIngredient_MODEL_GPT_3_5_XXX` -> `ModelEmbeddings_MODEL_GPT_3_5_XXX` ?